src/spectral.jl

```diff
@@ -133,3 +133,62 @@ function CombinatorialAdjacency(g::Graph)
     return CombinatorialAdjacency{Float64, typeof(g), typeof(d)}(g,d)
 end
 end # @require
+
+
+"""Returns a sparse node-arc incidence matrix for a graph, indexed by
+`[v, i]`, where `i` is in `1:ne(g)`, indexing an edge `e`. For
+directed graphs, a value of `-1` indicates that `src(e) == v`, while a
+value of `1` indicates that `dst(e) == v`. Otherwise, the value is
+`0`. For undirected graphs, both entries are `1`.
+"""
+function incidence_matrix(g::Graph, T::DataType=Int)
+    n_v = nv(g)
+    n_e = ne(g)
+    nz = 2 * n_e
+
+    # every col has the same 2 entries
+    u_coef, v_coef = one(T), one(T)
+    colpt = collect(1:2:(nz + 1))
+    nzval = repmat([u_coef, v_coef], n_e)
+
+    # iterate over edges for row indices
+    rowval = zeros(Int, nz)
+    i = 1
+    for u in vertices(g)
+        for v in out_neighbors(g, u)
+            if is_ordered(Edge(u, v)) # add every edge only once
+                rowval[2*i - 1] = u
```

Comment:
Question: why not combine these two methods and stick this check in the combined method, sort of like this?

Also, I'm taking back what I said about using `is_ordered` here. It's much less efficient to create an Edge and use `is_ordered()` than it is to just check `u < v`.

```
isdir = is_directed(g)
```
then, in this loop,
```
   isdir && u < v && ...
```